app/controllers/shared/api.php

```diff
@@ -114,6 +114,33 @@
 
     $deletes->setProject($project);
     $database->setProject($project);
+
+    $useCache = $route->getLabel('cache', false);
+    if ($useCache) {
+            $key = md5($request->getURI() . $request->getServer('query_string'));
+            $cache = new Cache(new Filesystem(APP_STORAGE_CACHE . DIRECTORY_SEPARATOR . 'app-' . $project->getId()));
+            $data = $cache->load($key, 60 * 60 * 24 * 30 * 1);
+        if (!empty($data)) {
+            $cacheLog = Authorization::skip(fn () => $dbForProject->getDocument('cache', $key));
+            if ($cacheLog->isEmpty()) {
+                Authorization::skip(fn () => $dbForProject->createDocument('cache', new Document([
+                        '$id' => $key,
+                        'accessedAt' => time(),
+                    ])));
+            } else {
+                $cacheLog->setAttribute('accessedAt', time());
+                Authorization::skip(fn () => $dbForProject->updateDocument('cache', $cacheLog->getId(), $cacheLog));
+            }
+
+            $data = json_decode($data, true);
+
+            $response
+                ->addHeader('Expires', \date('D, d M Y H:i:s', \time() + (60 * 60 * 24 * 30)) . ' GMT')
+                ->addHeader('X-Appwrite-Cache', 'hit')
+                ->setContentType($data['content-type'])
+                ->file(base64_decode($data['payload']), $data['content-type'], $data['date']);
```

Comment:
I have second thoughts about this `file()` method. It doesn't feel like it adds much value compared to just using `send()`. Anyway we need to send the content type one line up, so not sure there's a point for it.